activity_browser/logger.py

```diff
@@ -0,0 +1,98 @@
+from PySide2.QtCore import Slot
+import logging
+import os, time
+import random, string
+
+
+from .signals import signals
+class Logger(logging.Logger):
+    """
+    Creates two logging streams, one to the console (with >=INFORMATION logs)
+    the second to a file specified with the name argument (with >=WARNING logs)
+    and attaches these to a logger.
+
+    Returns the logging device
+    """
+    file_handler = None
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.setLevel(logging.INFO)
+
+        self.stream_handler = logging.StreamHandler()
+        self.stream_handler.setLevel(logging.INFO)
+        # Threads have their own Logger, but these refer to the sessions log file
+        if Logger.file_handler is None:
+            dir_path = os.getcwd() + "/.logs"
+            os.makedirs(dir_path, exist_ok=True)
+            Logger.cleanDirectory(dir_path)
```

Comment:
So you are right on these points, I've extended the period to a year. I've also provided a timestamp for the file names, although I haven't changed the stem of the name (ab_logs). This might not be required however, it might be useful to have some separation if people start going crazy with plugins